Crashlytics/Crashlytics/Models/Record/FIRCLSReportAdapter.m

```diff
@@ -77,21 +78,69 @@ - (void)loadMetaDataFile {
+/// uncaught exception crashed the app).
+- (void)loadAllCrashFiles {
+  BOOL hasInitializedCommonComponents = false;
+
+  for (NSString *crashFilePath in [FIRCLSInternalReport crashFileNames]) {
+    NSString *path = [self.folderPath stringByAppendingPathComponent:crashFilePath];
+
+    // Skip if the certain crash file doesn't exist
+    if (![[NSFileManager defaultManager] fileExistsAtPath:path]) {
+      continue;
+    }
+
+    NSDictionary *dicts = [FIRCLSReportAdapter combinedDictionariesFromFilePath:path];
+
+    NSDictionary *exceptionDict = dicts[@"exception"];
+    NSDictionary *machExceptionDict = dicts[@"mach_exception"];
+    NSDictionary *signalDict = dicts[@"signal"];
+
+    // These fields are specific to the type of crash file
+    if (exceptionDict) {
+      self.exception = [[FIRCLSRecordException alloc] initWithDict:exceptionDict];
+
+    } else if (machExceptionDict) {
+      self.mach_exception = [[FIRCLSRecordMachException alloc] initWithDict:machExceptionDict];
+
+    } else if (signalDict) {
+      self.signal = [[FIRCLSRecordSignal alloc] initWithDict:signalDict];
+    }
+
+    // These fields are common across all crash files. The order of precedence is
+    // Exception > Mach Exception > Signal. Since we are iterating in that order,
+    // once any of these fields have been, do not overwrite them.
+    if (hasInitializedCommonComponents) {
+      continue;
+    }
+
+    hasInitializedCommonComponents = true;
 
-  self.signal = [[FIRCLSRecordSignal alloc] initWithDict:dicts[@"signal"]];
-  self.runtime = [[FIRCLSRecordRuntime alloc] initWithDict:dicts[@"runtime"]];
-  self.processStats = [[FIRCLSRecordProcessStats alloc] initWithDict:dicts[@"process_stats"]];
-  self.storage = [[FIRCLSRecordStorage alloc] initWithDict:dicts[@"storage"]];
+    self.runtime = [[FIRCLSRecordRuntime alloc] initWithDict:dicts[@"runtime"]];
+    self.processStats = [[FIRCLSRecordProcessStats alloc] initWithDict:dicts[@"process_stats"]];
+    self.storage = [[FIRCLSRecordStorage alloc] initWithDict:dicts[@"storage"]];
 
-  // The thread's objc_selector_name is set with the runtime's info
-  self.threads = [FIRCLSRecordThread threadsFromDictionaries:dicts[@"threads"]
-                                                 threadNames:dicts[@"thread_names"]
-                                      withDispatchQueueNames:dicts[@"dispatch_queue_names"]
-                                                 withRuntime:self.runtime];
+    // The thread's objc_selector_name is set with the runtime's info
+    self.threads = [FIRCLSRecordThread threadsFromDictionaries:dicts[@"threads"]
+                                                   threadNames:dicts[@"thread_names"]
+                                        withDispatchQueueNames:dicts[@"dispatch_queue_names"]
+                                                   withRuntime:self.runtime];
+  }
+}
+
+// Reimplements Protobuf.scala#L102 (getCrash)
```

Comment:
Since this isn't a TODO, remove this line since it references to the scala source file.

Reply:
Talked offline